src/HydratorInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+declare(strict_types=1);
+
+namespace Yiisoft\Hydrator;
+
+/**
+ * @psalm-type MapType = array<string,string|list<string>>
+ */
+interface HydratorInterface
+{
+    /**
+     * @psalm-param MapType $map
+     */
+    public function populate(object $model, array $data = [], array $map = [], bool $strict = false): void;
```

Comment:
Maybe `hydrate` is better name?

Reply:
> Maybe `hydrate` is better name?

Maybe. Both names is OK for me.

> And merge these methods into one?

Merge is bad idea for me:
1) populate not need return result;
2) it's different cases.

Reply:
What are the different cases?

Reply:
Create and populate. Its different cases

Reply:
I don't think create is hydrator's task. Hydrator should only expect object instance not class string.

Reply:
In our case - create instance of object is task of hydrator also. For example:

```
final class PostModel {
	public function __construct(
		private string $name,
		private int $count,
	) {}
}

public function actionCreate(PostModel $model): array
{
	// ...
}
```

hydrator populate class properties and constructor parameters — is are same tasks.

Reply:
This package is just `hydrator` so should be clear, it seems create->hydrate case-specific hydrator. I would suggest create extended interface for this case.
```php
interface PopulatorInterface extends HydratorInterface
```

Reply:
`hydrate` in this package reuired `create` method, because supports such things:

```php
final class MyDto {
	public int $a;
	public NestedDto $dto;
}

final class NestedDto {
	public int $b;
}

$myDto = new MyDto();
$hydrator->populate($myDto, [
	'a' => 1,
	'dto' => ['b' => 2],
]);
```

On hydrate instance of `MyDto` need create new instance of `NestedDto`.